api/v1alpha/api.proto

```diff
@@ -37,6 +37,24 @@ enum ImageType {
         IMAGE_TYPE_OCI       = 3;
 }
 
+// InsecureOption defines the insecure option flags to be used.
+enum InsecureOption {
+        INSECURE_OPTION_NONE    = 0;
+        INSECURE_OPTION_ALL     = 1;
+        INSECURE_OPTION_IMAGE   = 2;
+        INSECURE_OPTION_TLS     = 4;
+        INSECURE_OPTION_ON_DISK = 8;
+        INSECURE_OPTION_HTTP    = 16;
+        INSECURE_OPTION_PUBKEY  = 32;
+}
+
+// DiscoverOption defines the discovery options to be used for fetching.
+enum DiscoverOption {
+        DISCOVER_OPTION_NONE       = 0;
```

Comment:
I propose FetchOptions, since these are parameters for a `Fetcher` or a `DistributionHandler` when #2964 will land, where it handles all the logic (use an image from the local store, fetch the missing blobs, fetch updated image blobs, update references #3071 etc...)